cel/library.go

```diff
@@ -240,7 +239,7 @@ func (l library) CompileOptions() []cel.EnvOption { //nolint:funlen,gocyclo
 					if !ok {
 						return types.Bool(false)
 					}
-					return types.Bool(l.validateURI(s, true))
+					return types.Bool(l.validateURI(s))
```

Comment:
Not usre if you'd rather these also be inlined as `types.Bool(newURI(s).uri()`, etc. Lmk if so.

Reply:
I'd prefer a `func isURI`, with the same doc comments as the source implementation. This makes porting easier because the `library` struct is out of the picture.

Alternatively, we can make all functions methods of `library`, but I don't see the point. They don't use state, and I think it's good that they don't.

Okay, `isEmail` uses state. But it doesn't have to:

```go
// See https://html.spec.whatwg.org/multipage/input.html#valid-e-mail-address
var emailRegexp = regexp.MustCompile("...")

// helpful comments
func isEmail(addr string) bool {
	return emailRegexp.MatchString(addr)
}
``` 

Whichever we do, it should be consistent.